src/dmd/escape.d

```diff
@@ -500,7 +500,8 @@ bool checkAssignEscape(Scope* sc, Expression e, bool gag)
                 if (!va.isScope() && inferScope)
                 {   //printf("inferring scope for %s\n", va.toChars());
                     va.storage_class |= STC.scope_ | STC.scopeinferred;
-                    va.storage_class |= v.storage_class & STC.return_;
+                    if (v.storage_class & STC.return_)
+                    va.storage_class |= STC.return_ | STC.returninferred;
```

Comment:
Bad indentation.

Doesn't this need a check for pre-existing `return` annotation?
```
if (!(va.storage_class & STC.return_))
    va.storage_class |= STC.return_ | STC.returninferred;
```
If `scope` and `return` can be inferred independently, the if-condition `!va.isScope()` seems to disallow that.